pkg/cloud/aws/actuators/machine/actuator.go

```diff
@@ -333,14 +333,19 @@ func (a *Actuator) CreateMachine(cluster *clusterv1.Cluster, machine *clusterv1.
 		},
 	}
 
+	clusterID, ok := getClusterID(machine)
```

Comment:
Personally i think using the `cluster` object for the cluster id is not correct, rather than the label on the machine object.

Is the cluster object getting dropped?

Reply:
Depends on if we want to assume `at most one cluster object per namespace`. @dgoodwin @csrwng are we fine with that assumption?

Reply:
@ingvagabund that assumption is currently done by the machine core controller https://github.com/kubernetes-sigs/cluster-api/blob/master/pkg/controller/machine/controller.go#L210:28 but I think that's a separate topic to discuss. We are already using `getClusterID` https://github.com/openshift/cluster-api-provider-aws/blob/master/pkg/cloud/aws/actuators/machine/utils.go#L118
In any case all this PR is doing is being consistent across the code and with our vision for the machine api based on my comment below https://github.com/openshift/cluster-api-provider-aws/pull/74#issuecomment-430359894